libraries/timeslider/src/components/TimeSlider/TimeSlider.tsx

```diff
@@ -0,0 +1,370 @@
+    ): Promise<void> => {
+        let start, end;
+        const promises: Promise<void>[] = [];
+        map.layers.forEach((tempLayer) => {
+            const layer = tempLayer as FeatureLayer;
+
+            promises.push(
+                layer.load().then(() => {
+                    if (layer.timeInfo) {
+                        if (
+                            start === undefined ||
+                            start > layer.timeInfo.fullTimeExtent.start
+                        ) {
+                            start = layer.timeInfo.fullTimeExtent.start;
+                        }
+                        if (
+                            end === undefined ||
+                            end < layer.timeInfo.fullTimeExtent.end
+                        ) {
+                            end = layer.timeInfo.fullTimeExtent.end;
+                        }
+                    }
+                })
+            );
+        });
+        // eslint-disable-next-line @typescript-eslint/no-floating-promises
+        return Promise.all(promises).then(() => {
+            widget.set("fullTimeExtent", new TimeExtent({ start, end }));
+            // Ensure time slider playback will end on the latest date in the full
+            // time extent.
+            if (
+                widget.effectiveStops &&
+                widget.effectiveStops[widget.effectiveStops.length - 1] <
+                    widget.fullTimeExtent.end
+            ) {
+                widget.effectiveStops.push(widget.fullTimeExtent.end);
+            }
+        });
+    };
+
+    const updateWidgetFromWebMapTimeSlider = (
+        widget: EsriTimeSlider,
+        timeSlider: __esri.WebMapTimeSlider
+    ): void => {
+        let timeExtentOption: string;
+        if (timeSlider.fullTimeExtent) {
+            widget.set("fullTimeExtent", timeSlider.fullTimeExtent);
+            widget.set("timeExtent", timeSlider.currentTimeExtent);
+        } else {
+            widget.set("fullTimeExtent", timeSlider.currentTimeExtent);
+            timeExtentOption = "currentTimeExtent";
+        }
+        // Set stops based on available properties in the time slider
+        // config in this order: StopByDate, StopByCount, StopByInterval
+        if (timeSlider.stops) {
+            widget.set("stops", {
+                dates: timeSlider.stops,
+                timeExtent: widget.fullTimeExtent,
+            });
+        } else if (timeSlider.numStops) {
+            widget.set("stops", {
+                count: timeSlider.numStops,
+                timeExtent: widget.fullTimeExtent,
+            });
+        } else if (timeSlider.stopInterval) {
+            widget.set("stops", {
+                interval: timeSlider.stopInterval,
+                timeExtent: widget.fullTimeExtent,
+            });
+        }
+        // Ensure time slider playback will end on the latest date in the full
+        // time extent.
+        if (
+            widget.effectiveStops &&
```

Comment:
This type of thing can be shortened with null chaining operator if you wanted, e.g. `widget.effectiveStops?.[widget.effectiveStops.length - 1] < ...`